kratos/geometries/geometry.h

```diff
@@ -3169,6 +3192,42 @@ class Geometry
 
     PointsArrayType mPoints;
 
+    ///@}
+    ///@name Id Bit-Change Operations
+    ///@{
+
+    /// Checks first bit in mId. 0 -> id; 1 -> name
+    bool IsGeometryIdString()
+    {
+        return mId & (IndexType(1) << 63);
+    }
+
+    inline bool IsGeometryIdString(IndexType Id) const
```

Comment:
The method name can be improved. like `IsIdGeneratedFromString`

Reply:
I agree. I changed it.